game-core/src/main/java/games/strategy/debug/error/reporting/StackTraceErrorReportFormatter.java

```diff
@@ -0,0 +1,107 @@
+   * string to take their place. At a minimum we will always be logging the location of the logging or
+   * exception, and we expect for there to always either be at least a log message or an exception.
+   */
+  private static String createTitle(final LogRecord logRecord) {
+
+    // an exception is optional, eg: 'log.severe("message only, no exception")'
+    final String exceptionNamePrefix = Optional.ofNullable(logRecord.getThrown())
+        .map(Throwable::getClass)
+        .map(Class::getSimpleName)
+        .map(simpleName -> simpleName + " - ")
+        .orElse("");
+
+    final String classShortName = logRecord.getSourceClassName().contains(".")
+        ? logRecord.getSourceClassName().substring(logRecord.getSourceClassName().lastIndexOf('.') + 1)
+        : logRecord.getSourceClassName();
+
+    final String sourceLocation = classShortName + "." + logRecord.getSourceMethodName();
+
+    // for title prefer to use the exception message when we have it, if not fall back to the log message
+    final String message = Optional.ofNullable(
+        Optional.ofNullable(logRecord.getThrown())
+            .map(Throwable::getMessage)
+            .orElse(logRecord.getMessage()))
+        .map(m -> ": " + m)
+        .orElse("");
+
+    return exceptionNamePrefix + sourceLocation + message;
+  }
+
+  /**
+   * Creates a string formatted in markdown. If we have an exception we'll log the full exception name and stack trace.
+   * If there is a log message we will log it, this might be redundant to the exception message (but that is okay!).
+   * Otherwise we will log base information like OS, TripleA version and Java version.
+   */
+  private String buildBody(@Nullable final String userDescription, final LogRecord logRecord) {
+    return Optional.ofNullable(Strings.emptyToNull(userDescription))
+        .map(
+            descripton -> "## User Description\n"
+                + descripton + "\n\n")
+        .orElse("")
+        + Optional.ofNullable(logRecord.getMessage())
+            .map(msg -> "## Log Message\n" + msg + "\n\n")
+            .orElse("")
+        + Optional.ofNullable(logRecord.getThrown())
+            .map(StackTraceErrorReportFormatter::throwableToString)
+            .orElse("")
+        + "## TripleA Version\n"
+        + ClientContext.engineVersion().toStringFull() + "\n\n"
+        + "## Java Version\n"
+        + SystemProperties.getJavaVersion() + "\n\n"
+        + "## Operating System\n"
+        + SystemProperties.getOperatingSystem();
+  }
+
+  private static String throwableToString(final Throwable e) {
+    return "## Exception \n"
+        + e.getClass().getName()
+        + Optional.ofNullable(e.getMessage())
+            .map(msg -> ": " + msg)
+            .orElse("")
+        + "\n\n"
+        + Optional.ofNullable(e.getStackTrace())
+            .map(
+                trace -> "## Stack Trace\n"
+                    + String.join("\n", stackTraceToString(trace))
```

Comment:
You should inline this join into the helper method, there's a reason `Collectors#joining` exists.

Reply:
@RoiEXLab I appreciate the comment and desire to improve the code, but the way it was delivered did strike me in absolutely the wrong way.

Specifically the "You should" and "there's a reason"

Personally my first response to the "you should" statement is "should I really", and I often wonder if the comment is myopic or even on-point with the right fix. Regardless, it does not seem to be a good CR practice in general to have such a tone.  I very much recommend this article: https://medium.freecodecamp.org/unlearning-toxic-behaviors-in-a-code-review-culture-b7c295452a3c

Notably:

> Helpful Behavior #1: use questions or recommendations to drive dialog

> Never make demands for people to implement changes or ask judgmental questions, because this does not open up a dialog between you and the person whose code you are reviewing.

> Unhelpful behavior #5: being sarcastic

"there's a reason Collectors#joining exists"

-------

I think given we are just going over syntax nit, giving an exact code example of the simplification would be most helpful as a reviewer and is what I would have preferred to have received (otherwise let it slide).

Reply:
You are right. I should've written that in another way. I want to apologise for that.
I think I'm just so used to knowing people I review code from in person, so that's probably where the bad habit comes from.
I'll try to keep that in mind in the future 👍

Reply:
Thanks! Water under the bridge, I've been in exactly the same shoes and had some hard lessons about it personally. I know / appreciate how much effort it is to review code, so I'm really happy to have that.

Reply:
`Collectors#joining` simplification added:  36fe92f97